patches/minecraft/net/minecraft/world/item/enchantment/EnchantmentHelper.java.patch

```diff
@@ -1,5 +1,31 @@
 --- a/net/minecraft/world/item/enchantment/EnchantmentHelper.java
 +++ b/net/minecraft/world/item/enchantment/EnchantmentHelper.java
+@@ -59,7 +_,12 @@
+       return Registry.f_122825_.m_7981_(p_182433_);
+    }
+ 
++   @Deprecated // forge: use getTagEnchantmentLevel or ItemStack::getEnchantmentLevel
+    public static int m_44843_(Enchantment p_44844_, ItemStack p_44845_) {
++      return p_44845_.getEnchantmentLevel(p_44844_);
++   }
++
++   public static int getTagEnchantmentLevel(Enchantment p_44844_, ItemStack p_44845_) {
```

Comment:
Just pointing this out so it isn't missed but I am unsure if this counts as a logical breaking change or not as there are times when a mod may be calling this but only want the tag enchantment levels (though it would be a lot more maintenance heavy to patch it a different way). Mostly just want more opinions on this given cases where even in vanilla (the grindstone menu) it needed to be changed to be explicitly NBT based.

Reply:
I agree its probably a breaking change, but I suspect it will affect minimal mods. Mods that implement custom enchantments shouldn't have any trouble, while mods that add custom enchanting methods or machines to remove enchantments will need to update to the tag based methods. I suspect even among custom enchantment methods, most of them are probably using the map method making it a very small list of mods affected, so my opinion is an announcement would be sufficient.

If that is deemed as too big of a breaking change, sounds like 1.18.2 is coming soon, so could always include it there.